### PR TITLE
Fix recursion in sync routine.

### PR DESCRIPTION
**Pull request by nerrad**
## Description

While working on #13716 I discovered a bug where when a control returns the results with an object that has a `type` property, `redux-routine` was recursing into that returned object treating it as an action.

When I brought this up in [slack](https://wordpress.slack.com/archives/C02QB2JS7/p1549904073023400), @youknowriad suggested it could be a bug in `redux-routine` and I tried his suggested fix and that seemed to work so this is the pull applying it.

## How has this been tested?

* [x] existing unit tests pass
* [x] e2e tests pass
* [x] Fixes the bugs reported in #13716
* [x] Testing against a plugin I've written that uses redux-routine to verify nothing is borked there.

## Types of changes

As far as I can tell, this is a non-breaking change.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
It would be good to add a unit test 🤷‍♂️

**Comment by nerrad** [comment]
> It would be good to add a unit test 🤷‍♂️

Hmm ya, I guess it would :) lemme see if I can write one that catches the bug.

**Comment by nerrad** [comment]
Added a unit test that fails without the fix in this branch and passes with the fix in this branch.

File: packages/redux-routine/src/test/index.js [hunk 1]
```diff
@@ -143,4 +143,21 @@ describe( 'createMiddleware', () => {
 
 		expect( store.getState() ).toBe( 2 );
 	} );
+
+	it( 'does not recurse when action like object returns from a sync ' +
+		'control', () => {
```
Comment (aduth) [inline packages/redux-routine/src/test/index.js]: Is the concatenation here something automated? Or are you especially strict with line length? 😄 

I don't have any strong feelings on it, but it made it slightly more difficult for me to read the test case (or at least, it came as a surprise to me in reading the label).

Comment (nerrad) [reply packages/redux-routine/src/test/index.js]: Of the various options I've played around with for handling line length issues, this is the one I've landed on as the one I can be most consistent with and also indents nicely.

Comment (aduth) [reply packages/redux-routine/src/test/index.js]: Oh, I can appreciate it. I'm of a similar obsession, particularly for comments, even to the point of enabling the visual "line" for my editor.

image

I didn't intend to make a thing of it. My only worry might be that the indentation implies that it's of a similar level as the body of the callback, when in fact it's still part of the earlier (`it`) level.

Comment (nerrad) [reply packages/redux-routine/src/test/index.js]: > I'm of a similar obsession, particularly for comments, even to the point of enabling the visual "line" for my editor.

Heh!  I do the same in my IDE + have my IDE configured to show me when line length is exceeded.

**Pull request merged**